core/vgmgr/virtual_group_manager.go

```diff
@@ -49,6 +50,25 @@ type GenerateGVGSecondarySPsPolicy interface {
 	GenerateGVGSecondarySPs() ([]uint32, error)
 }
 
+// ExcludeFilter applies on an ID to check if it should be excluded
+type ExcludeFilter interface {
+	Apply(id uint32) bool
+}
+
+type ExcludeIDFilter struct {
+	ExcludeIDs map[uint32]struct{}
+}
+
+func NewExcludeIDFilter(ids IDSet) ExcludeFilter {
+	return &ExcludeIDFilter{
+		ExcludeIDs: ids,
+	}
+}
+func (f *ExcludeIDFilter) Apply(id uint32) bool {
```

Comment:
More comments can be added. BTW, the semantics of filter is generally to filter out entities that meet the conditions, here is the opposite semantics, there is a certain cost of understanding.

Reply:
added